network-api/networkapi/wagtailpages/factory/youtube_regrets_page.py

```diff
@@ -30,6 +34,25 @@ class Meta:
     regret_stories = Faker("streamfield", fields=["regret_story"] * 28)
 
 
+class AppInstallPageFactory(PageFactory):
+    class Meta:
+        model = AppInstallPage
+        exclude = (
+            "title_text",
+            "header_text",
+            "header",
+        )
+
+    title = "Regrets Reporter Page"
+    slug = "regretsreporter"
+    hero_heading = Faker("text", max_nb_chars=50)
+    hero_subheading = Faker("text", max_nb_chars=50)
+    hero_background = SubFactory(image_factory.ImageFactory)
```

Comment:
```suggestion
    hero_background = SubFactory(image_factory.ImageFactory)
    download_buttons = Faker("streamfield", fields=["app_install_download_button"] * 2)
```


Since `AppInstallPages` usually have link buttons directing to download pages (like how [YoutubeRegrets](https://foundation.mozilla.org/en/youtube/regretsreporter/) has "Add To Firefox" and "Add to Chrome" buttons), it would be great if we could generate those through the factory so users and devs can see the full potential of the page.

We can add these buttons by including the line above ^ in the factory and making the following updates to `streamfield_provider.py`:

1. Add a `generate_app_install_download_button_field` method:
```python
def generate_app_install_download_button_field():

    return generate_field(
        "button",
        {
            "link_to": "external_url",
            "external_url": fake.url(schemes=["https"]),
            "label": " ".join(fake.words(nb=2)),
            "new_window": True,
        },
    )
```

2. Add the method to the list of valid fields within the `StreamfieldProvider`:
```python
    def streamfield(self, fields=None):
        """
        Generate a streamfield string containing the fields optionally defined in fields.
        Defaults to ['header', 'paragraph']
        """

        valid_fields = {
            "app_install_download_button": generate_app_install_download_button_field,
            ...
        }
```

Doing so will generate two dummy buttons that look like this 👍 : 

![Screenshot 2024-07-08 at 13-22-20 Regrets Reporter Page - Mozilla Foundation](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/903a7859-8536-44ef-8864-785e05f0a271)